### DISCUSSION TITLE
AssemblyScript Contract Examples & Issues

### DISCUSSION THREAD

**Discussion by christian-rogobete** [stellar/sorobanathon#25] [General]
Hello, I created a first, "basic" version of the  [AssemblyScript Soroban SDK](https://github.com/Soneso/as-soroban-sdk). 

In this Discussion, I would like to submit a couple of working **example contracts**, that I created using the new AS SDK:

https://github.com/Soneso/as-soroban-examples

Also, I would like to point out a few issues I've run into, when developing with AS and running the ```wasm``` module in the soroban vm.  

Basic problem  num. 1 -  I can not add strings: 

```typescript
let a = "b" + "c";
```

when running in the ```soroban-cli``` I get following error:

```text
error: HostError
Value: Status(VmError(Validation))

Debug events (newest first):
   0: "Validation"
```

The issue example and instructions on how to reproduce can be found here:
https://github.com/Soneso/as-soroban-examples/tree/main/some_issues

**Comment by achallagundla** [comment]
Thanks for this! Looking forward to reviewing it and get back to you soon!

**Comment by rice2000** [comment]
Thanks for this!

**Comment by leighmcculloch** [comment]
This is so fantastic. Very nice. 

Regarding the adding strings issue, this is probably because of a lack of allocator, and the wasm environment in Soroban doesn't provide any of the functions that AS expects: https://www.assemblyscript.org/runtime.html#interface. Do you think that's correct, or do you think something else going on here?

@christian-rogobete If you're up for it it'd be great if you could open an issue for this problem on the `stellar/rs-soroban-env` GitHub repository here with as much detail as possible for reproducing: https://github.com/stellar/rs-soroban-env/issues/new

**Reply by christian-rogobete** [reply]
Thank you! :)
I will add an issue on the ```rs-soroban-env``` GitHub repository for this problem.

**Reply by christian-rogobete** [reply]
added the new issue here:
https://github.com/stellar/rs-soroban-env/issues/545

**Comment by achallagundla** [comment]
Heya Christian,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 3000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately. 

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

**Reply by christian-rogobete** [reply]
Heya Anuhya,

sounds great! Thank you very much! I filled the form. 

All the best and have a great week!
Christian
